service/forward_service.cc

```diff
@@ -353,6 +353,13 @@ future<query::forward_result> forward_service::dispatch_to_shards(
                 result = r;
             }
         }
+
+        query::forward_result::printer res_printer{
+            .functions = get_functions(req),
```

Comment:
Is this get_functions slow? If so consider using seastar::value_of().

Reply:
Probably it's slow, in particular it may iterate over declared function. Wrapped all of the uses of `forward_result::printer` into `seastar::value_of`.